compiler/rustc_hir_analysis/messages.ftl

```diff
@@ -178,3 +178,14 @@ hir_analysis_invalid_union_field =
 
 hir_analysis_invalid_union_field_sugg =
     wrap the field type in `ManuallyDrop<...>`
+
+hir_analysis_return_type_notation_on_non_rpitit =
+    return type notation used on function that is not `async` and does not return `impl Trait`
+    .note = function returns `{$ty}`, which is not compatible with associated type return bounds
+    .label = this function must be `async` or return `impl Trait`
```

Comment:
For completeness, I think it'd be nice to support this case someday, but I don't think it's very useful in practice and I don't think we need or should implement it right now.